src/sagemaker/session.py

```diff
@@ -1225,6 +1225,115 @@ def logs_for_job(self, job_name, wait=False, poll=10):  # noqa: C901 - suppress
             billable_time = (description['TrainingEndTime'] - description['TrainingStartTime']) * instance_count
             print('Billable seconds:', int(billable_time.total_seconds()) + 1)
 
+    def logs_for_transform_job(self, job_name, wait=False, poll=10):  # noqa: C901 - suppress complexity warning
+        """Display the logs for a given transform job, optionally tailing them until the
+        job is complete. If the output is a tty or a Jupyter cell, it will be color-coded
+        based on which instance the log entry is from.
+        Args:
+            job_name (str): Name of the transform job to display the logs for.
+            wait (bool): Whether to keep looking for new log entries until the job completes (default: False).
+            poll (int): The interval in seconds between polling for new log entries and job completion (default: 5).
+        Raises:
+            ValueError: If waiting and the transform job fails.
+        """
+
+        description = self.sagemaker_client.describe_transform_job(TransformJobName=job_name)
+        instance_count = description['TransformResources']['InstanceCount']
+        status = description['TransformJobStatus']
+
+        stream_names = []  # The list of log streams
```

Comment:
is there enough common logic between this method and `logs_for_job` that could be refactored into a helper method?

Reply:
refactored the two methods...

Reply:
Agreed with @laurenyu here. It seems that you can refactor this even more, only the describe job call seems to be different here.

Reply:
Most of this function still is a duplication of `logs_for_job`. Let's refactored the common logic in a private method.